### DISCUSSION TITLE
Hermite element in MOOSE

### DISCUSSION THREAD

**Discussion by ZhouXD1989** [idaholab/moose#16391] [Q&A General] [answered]
Hello all,

I am trying to develop an element to deal with C1 continuity problem in moose. Beam element with Hermite interpolation seems to be a good example for me. However, I checked the beam element in moose, it uses the Lagrange interpolation for both displacements and rotations, as shown in the following:
https://mooseframework.inl.gov/modules/tensor_mechanics/beam_vandv.html
Why is the beam element different from the commonly used beam element? Is there a reference for this beam element?
Is there an element in moose with Hermite interpolation?

Thank you very much!

Best wishes,
Xiandong

**Comment by jiangwen84** [comment, marked as answer]
I can partially answer your question. If I remember correctly, MOOSE's Hermite shape function does not work with displaced mesh. A displaced mesh (current configuration) is needed for large deformation problem. Like what you found,  the current implementation of beam element in MOOSE separates displacement and rotation variables, which allows for using C0 Lagrange shape function.  This is basically a mixed/split formulation of a high-order PDE. Other examples can be found in MOOSE to solve 4th order Cahn-Hilliard equation in both split and total formulation (use Hermite).

**Comment by ZhouXD1989** [comment]
Dear Wen Jiang,

I tried the total formulation with Hermite element. It works well. Thanks a
lot!

Best wishes,
Xiandong

Wen Jiang <notifications@github.com> 于2020年12月7日周一 下午8:49写道：

> I can partially answer your question. If I remember correctly, MOOSE's
> Hermite shape function does not work with displaced mesh. A displaced mesh
> (current configuration) is needed for large deformation problem. Like what
> you found, the current implementation of beam element in MOOSE separates
> displacement and rotation variables, which allows for using C0 Lagrange
> shape function. This is basically a mixed/split formulation of a high-order
> PDE. Other examples can be found in MOOSE to solve 4th order Cahn-Hilliard
> equation in both split and total formulation (use Hermite).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/discussions/16391#discussioncomment-162267>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALDXGIFD2YTKTZPNIIWUDYLSTUWUBANCNFSM4UNZAXCA>
> .
>


-- 
Best regards,
Xiandong Zhou
Mechanics of Functional Materials, Institution of Materials Science, TU
Darmstadt, Germany.
E-mail: zhouxd22@g <taoyong@pku.edu.cn>mail.com

**Reply by andrey1887** [reply]
Hello, @ZhouXD1989.  I currently try to implement Euler-Bernuolli beam with Hermite shape functions too. Can you tell me please, what's the difference compared to a C0 Timoshenko beam. I have changed `computeStiffnessMatrix`, but i can't understand how to pass it into `local_ke` in `StressDivergenceBeam::computeJacobian`.  

Variables in input file looks next:
```
[Variables]

  [disp_x]
    order = FIRST
    family = LAGRANGE
  []
  [disp_y]
    order = THIRD
    family = HERMITE
  []
  [disp_z]
    order = THIRD
    family = HERMITE
  []
  [rot_x]
    order = FIRST
    family = LAGRANGE
  []
  [rot_y]
    order = THIRD
    family = HERMITE
  []
  [rot_z]
    order = THIRD
    family = HERMITE
  []
[]
```
Is that right?
